# User-Studies 

In [11]:
import pandas as pd
import scipy.stats as stats
from statsmodels.stats.multicomp import pairwise_tukeyhsd

## Data

In [21]:
df = pd.read_csv("Data/IN5060_fall24-Pre-study.csv")

# reading all the files
pre_study = pd.read_csv('Data/IN5060_fall24-pre-study.csv')
during_study = pd.read_csv("Data/IN5060_fall24-during-study.csv")
post_study = pd.read_csv('Data/IN5060_fall24-post-study.csv')

# Display all columns and expand display width
pd.set_option('display.max_columns', None)     # Show all columns
pd.set_option('display.width', 1000)           # Set the display width to fit large tables
pd.set_option('display.max_colwidth', None)    # Show full column content without truncation
pd.set_option('display.max_rows', None)        # Show all rows (remove this if there are too many rows)

print(df.head(10))

                      Timestamp Participant ID (top-right corner of the screen)  How old are you?            Gender?          What describes you the best?  How comfortable are you in expressing emotions? Have you ever played with the "drummer bot" dB before?  How much prior experience do you have with interactive musical interfaces (e.g. sensor systems, digital control devices)?  How often do you use AI tools for creativity (e.g., writing, coding, music, fine arts, food recipes, etc.)? Please indicate if you have any of the following health problems.
0   2024/11/04 5:15:10 PM GMT+1                                            test                66  Prefer not to say    Student/semi-professional musician                                                3                                                    Yes                                                                                                                          4                                                              

# Pre Study

# During Study

In [ ]:
# Perform ANOVA on "I felt excited" across different "Session ID" groups
anova_results = stats.f_oneway(
    *[during_study[during_study['Session ID (the last session that the participant has completed)'] == session]['I felt excited']
      for session in during_study['Session ID (the last session that the participant has completed)'].unique()]
)

print("F-statistic:", anova_results.statistic)
print("p-value:", anova_results.pvalue)

F-statistic: 1.7634861987486083
p-value: 0.00657328546840777


The ANOVA test results show an F-statistic of approximately 1.76 and a p-value of 0.0066. Since the p-value is below the typical significance level (e.g., 0.05), we can conclude that there are statistically significant differences in excitement levels across the different session types.

In [ ]:
# Prepare data for Tukey's HSD test
session_data = during_study[['Session ID (the last session that the participant has completed)', 'I felt excited']].dropna()
session_data.columns = ['SessionID', 'Excitement']

tukey_results = pairwise_tukeyhsd(endog=session_data['Excitement'], groups=session_data['SessionID'], alpha=0.05)

tukey_summary = tukey_results.summary()

# Convert the summary to a pandas DataFrame for easier viewing
tukey_df = pd.DataFrame(data=tukey_summary.data[1:], columns=tukey_summary.data[0])

print(tukey_df)

# Display the results
#print(tukey_results.summary())

           group1        group2  meandiff   p-adj   lower   upper  reject
0           ALICE       VOYAGER   -1.5000  1.0000 -6.8125  3.8125   False
1           ALICE             1   -0.8750  1.0000 -4.3042  2.5542   False
2           ALICE             2   -1.2143  1.0000 -4.6922  2.2636   False
3           ALICE             3   -1.5000  0.9992 -4.8909  1.8909   False
4           ALICE             4   -2.3333  0.8124 -5.8750  1.2084   False
5           ALICE         ALICE   -0.9881  1.0000 -4.0916  2.1154   False
6           ALICE        ALLICE    0.5000  1.0000 -3.8377  4.8377   False
7           ALICE         Alice    0.2500  1.0000 -3.5065  4.0065   False
8           ALICE          CYSP   -1.1875  1.0000 -4.2865  1.9115   False
9           ALICE         Crysp    0.5000  1.0000 -4.8125  5.8125   False
10          ALICE          Cysp    1.0000  1.0000 -3.3377  5.3377   False
11          ALICE         G17P1    0.5000  1.0000 -4.8125  5.8125   False
12          ALICE       Grugru2   -1.5

# Post Study